assignments/array-methods.js

```diff
@@ -58,26 +58,62 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 let fullName = [];
 console.log(fullName);
 
+runners.forEach((getNames)=>{
+    let names = getNames.first_name + " " + getNames.last_name;
+    fullName.push(names)
+  });
+
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
 console.log(allCaps); 
 
+caps.push(capitalize.first_name.toUpperCase())
+const caps = runners.map((capitalize));{
+  return capitalize.first_name.toUpperCase()
+}
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(runners => runners.shirt_size === "L");
 console.log(largeShirts);
 
+
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+runners.forEach(function (total) {
+    let tickettotal = total.donation
+    ticketPriceTotal.push(tickettotal);
+});
+ticketPriceTotal = ticketPriceTotal.reduce(function (accumulator, currentValue) {
+    return accumulator + currentValue
+})
+
+ console.log(ticketPriceTotal);	
```

Comment:
For challenge 4, you created a brand new array and pushed just the donations into it. From there, you could reduce that array into a sum. That is well done.

Just for funsies, remember that `map` also creates a new array, pushes values specified by the callback function into it, and returns it all on its own, without you needing to make the new array and push it to yourself. Therefore, you could this:

```js
const donations = runners.map(runner => runner.donation);
```

That will give you an array of all the donations. You can then say:

```js
const ticketPriceTotal = donations.reduce(function (accumulator, currentValue) {
    return accumulator + currentValue;
});
```

And that would get you to the same place.